addons/account/models/account_move.py

```diff
@@ -3339,22 +3347,15 @@ def _reverse_moves(self, default_values_list=None, cancel=False):
 
         reverse_moves = self.env['account.move']
         for move, default_values in zip(self, default_values_list):
-            default_values.update({
-                'move_type': TYPE_REVERSE_MAP[move.move_type],
-                'reversed_entry_id': move.id,
-                'partner_id': move.partner_id.id,
-            })
+            move._prepare_reversal_values(default_values)
             reverse_moves += move.with_context(
                 move_reverse_cancel=cancel,
                 include_business_fields=True,
                 skip_invoice_sync=move.move_type == 'entry',
             ).copy(default_values)
 
         reverse_moves.with_context(skip_invoice_sync=cancel).write({'line_ids': [
-            Command.update(line.id, {
-                'balance': -line.balance,
-                'amount_currency': -line.amount_currency,
-            })
+            Command.update(line.id, line._prepare_reversal_values())
```

Comment:
Why would you need anything in here?
This only applies to misc operations, where the only thing that needs to be done is reverse the amounts; everything else needs to be exactly the same.